pkg/pillar/evetpm/tpm.go

```diff
@@ -802,6 +853,112 @@ func pcrBankSHA256EnabledHelper() bool {
 	return err == nil
 }
 
+func getTpmEventLogFiles(evtDir string, evtFile string) ([]string, error) {
+	files, err := filepath.Glob(evtDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to enumerate tpm(s)")
+	}
+
+	tpmEvtFiles := make([]string, 0)
+	for _, path := range files {
+		evtBin := filepath.Join(path, evtFile)
+		if fileutils.FileExists(nil, evtBin) {
+			tpmEvtFiles = append(tpmEvtFiles, evtBin)
+		}
+	}
+
+	return tpmEvtFiles, nil
+}
+
+func collectPrevTpmEventLogs(evtLogSealing string, evtLogFailing string) error {
+	// just call it to count available tpm devices
+	paths, err := getTpmEventLogFiles(TPMEvtLogDir, TpmEvtLogBinFile)
+	if err != nil || len(paths) == 0 {
+		return fmt.Errorf("failed to find any TPM event log file: %w", err)
+	} else if len(paths) == 0 {
+		return fmt.Errorf("failed to find any TPM event log file")
+	}
+
+	saveFailed := len(paths)
+	for i := range paths {
+		sealSuc := fmt.Sprintf(TpmEvtLogSavePattern, evtLogSealing, i)
+		sealFail := fmt.Sprintf(TpmEvtLogSavePattern, evtLogFailing, i)
+		if fileutils.FileExists(nil, sealSuc) && fileutils.FileExists(nil, sealFail) {
+			prevSealSuc := fmt.Sprintf(TpmEvtLogCollectPattern, sealSuc)
+			prevSealFail := fmt.Sprintf(TpmEvtLogCollectPattern, sealFail)
+			if err := os.Rename(sealSuc, prevSealSuc); err != nil {
+				fmt.Fprintf(os.Stderr, "failed to collect tpm%d previous (success) event log: %v", i, err)
+				continue
+			}
+			if err := os.Rename(sealFail, prevSealFail); err != nil {
+				fmt.Fprintf(os.Stderr, "failed to collect tpm%d previous (fail) event log: %v", i, err)
+				_ = os.Rename(prevSealSuc, sealSuc)
+				continue
+			}
+		}
+
+		saveFailed--
+	}
+
+	if saveFailed != 0 {
+		return fmt.Errorf("failed to collect %d number of previous TPM event logs", saveFailed)
+	}
+
+	return nil
+}
+
+func cleanSavedTpmEventLogs(evtLogSealing string, evtLogFailing string) error {
```

Comment:
I thought this was cleaning/removing stuff from the content of the logs, and not deleting the old files.